Sources/GRPCCore/RPCError.swift

```diff
@@ -35,18 +35,62 @@ public struct RPCError: Sendable, Hashable, Error {
   /// The original error which led to this error being thrown.
   public var cause: (any Error)?
 
-  /// Create a new RPC error.
+  /// Create a new RPC error. If the given `cause` is also an ``RPCError`` sharing the same `code`,
+  /// then they will be flattened into a single error, by merging the messages and metadata.
   ///
   /// - Parameters:
   ///   - code: The status code.
   ///   - message: A message providing additional context about the code.
   ///   - metadata: Any metadata to attach to the error.
   ///   - cause: An underlying error which led to this error being thrown.
-  public init(code: Code, message: String, metadata: Metadata = [:], cause: (any Error)? = nil) {
-    self.code = code
-    self.message = message
-    self.metadata = metadata
-    self.cause = cause
+  public init(
+    code: Code,
+    message: String,
+    metadata: Metadata = [:],
+    cause: (any Error)? = nil
+  ) {
+    if let rpcErrorCause = cause as? RPCError, rpcErrorCause.code == code {
+      self.code = code
+      self.message = message + " \(rpcErrorCause.message)"
+      var mergedMetadata = metadata
+      mergedMetadata.add(contentsOf: rpcErrorCause.metadata)
+      self.metadata = mergedMetadata
+      self.cause = rpcErrorCause.cause
+    } else {
```

Comment:
Can't we just use the other `init` here rather than duplicating this?
```swift
    if let rpcErrorCause = cause as? RPCError {
      self = .init(...)
    } else {
```